### ISSUE TITLE
Xournal++ crashes on startup in Wayland (Signal 6)

### ISSUE THREAD

**Issue by RandomSpaceship** [xournalpp/xournalpp#5691]
### Operating System

Linux

### (Linux only) Distribution

NixOS

### (Linux only) Desktop Environment

KDE Plasma

### (Linux Only) Display Server

Wayland

### Installation Method

home-manager

### Xournal++ Version

1.2.3

### libgtk Version

3.24.41

### Bug Description

Upon starting Xournal++ in Wayland, it crashes before a window appears. After switching back to X11, it runs normally (and prompts to open a bug report).

### Expected Behaviour

The app should open.

### Steps to Reproduce

1. Log out
2. Change desktop selection from lockscreen to KDE (Wayland)
3. Log in
4. Attempt to launch Xournal++

### Additional Context

I'm using a laptop with a touchscreen, if that makes any difference. In X11, the app works perfectly.

Error log:
[errorlog.20240517-151326.log](https://github.com/xournalpp/xournalpp/files/15343938/errorlog.20240517-151326.log)
```
Date: Fri May 17 15:13:26 2024
Error: signal 6
Xournal++ version 1.2.3
Gtk version 3.24.41

[bt]: (0) /etc/profiles/per-user/james/bin/xournalpp() [0x4d08af]
[bt]: (1) /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(+0x3ff30) [0x7fc982a52f30]
[bt]: (2) /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(+0x8fefc) [0x7fc982aa2efc]
[bt]: (3) /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(gsignal+0x16) [0x7fc982a52e86]
[bt]: (4) /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(abort+0xd7) [0x7fc982a3b935]
[bt]: (5) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libglib-2.0.so.0(+0x23015) [0x7fc983281015]
[bt]: (6) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libglib-2.0.so.0(g_assertion_message_error+0xb1) [0x7fc9832e9d91]
[bt]: (7) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1cb850) [0x7fc9839cb850]
[bt]: (8) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1cbd27) [0x7fc9839cbd27]
[bt]: (9) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1cc03e) [0x7fc9839cc03e]
[bt]: (10) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1e019d) [0x7fc9839e019d]
[bt]: (11) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
[bt]: (12) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
[bt]: (13) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1e08c3) [0x7fc9839e08c3]
[bt]: (14) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4863) [0x7fc983aa4863]
[bt]: (15) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
[bt]: (16) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
[bt]: (17) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
[bt]: (18) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xe74d3) [0x7fc9838e74d3]
[bt]: (19) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4863) [0x7fc983aa4863]
[bt]: (20) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
[bt]: (21) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xd9c48) [0x7fc9838d9c48]
[bt]: (22) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
[bt]: (23) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
[bt]: (24) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xd8fa3) [0x7fc9838d8fa3]
[bt]: (25) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4863) [0x7fc983aa4863]
[bt]: (26) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height+0x5c) [0x7fc983aa407c]
[bt]: (27) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4746) [0x7fc983aa4746]
[bt]: (28) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
[bt]: (29) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a507b) [0x7fc983aa507b]
[bt]: (30) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x30404c) [0x7fc983b0404c]
[bt]: (31) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
[bt]: (32) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
[bt]: (33) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x302466) [0x7fc983b02466]
[bt]: (34) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4746) [0x7fc983aa4746]
[bt]: (35) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
[bt]: (36) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xd9c48) [0x7fc9838d9c48]
[bt]: (37) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
[bt]: (38) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
[bt]: (39) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xd8fa3) [0x7fc9838d8fa3]
[bt]: (40) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4863) [0x7fc983aa4863]
[bt]: (41) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height+0x5c) [0x7fc983aa407c]
[bt]: (42) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xd4391) [0x7fc9838d4391]
[bt]: (43) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4746) [0x7fc983aa4746]
[bt]: (44) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height+0x5c) [0x7fc983aa407c]
[bt]: (45) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x36b71a) [0x7fc983b6b71a]
[bt]: (46) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xcc93c) [0x7fc9838cc93c]
[bt]: (47) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4746) [0x7fc983aa4746]
[bt]: (48) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
[bt]: (49) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a507b) [0x7fc983aa507b]
[bt]: (50) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x36d5e4) [0x7fc983b6d5e4]
[bt]: (51) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x371c3a) [0x7fc983b71c3a]
[bt]: (52) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xccd5f) [0x7fc9838ccd5f]
[bt]: (53) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_closure_invoke+0x138) [0x7fc9833c0668]
[bt]: (54) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(+0x2bc80) [0x7fc9833d4c80]
[bt]: (55) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(+0x2d4d1) [0x7fc9833d64d1]
[bt]: (56) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_signal_emit_valist+0x32) [0x7fc9833dc232]
[bt]: (57) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_signal_emit+0x8f) [0x7fc9833dc2df]
[bt]: (58) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_realize+0xc8) [0x7fc983b61ee8]
[bt]: (59) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x36fa72) [0x7fc983b6fa72]
[bt]: (60) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_closure_invoke+0x138) [0x7fc9833c0668]
[bt]: (61) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(+0x2bc80) [0x7fc9833d4c80]
[bt]: (62) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(+0x2d4d1) [0x7fc9833d64d1]
[bt]: (63) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_signal_emit_valist+0x32) [0x7fc9833dc232]
[bt]: (64) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_signal_emit+0x8f) [0x7fc9833dc2df]
[bt]: (65) /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_show+0xaa) [0x7fc983b5bb8a]
[bt]: (66) /etc/profiles/per-user/james/bin/xournalpp() [0x4d95d9]
[bt]: (67) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_closure_invoke+0x138) [0x7fc9833c0668]
[bt]: (68) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(+0x2bb3c) [0x7fc9833d4b3c]
[bt]: (69) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(+0x2d4d1) [0x7fc9833d64d1]
[bt]: (70) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_signal_emit_valist+0x32) [0x7fc9833dc232]
[bt]: (71) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgobject-2.0.so.0(g_signal_emit+0x8f) [0x7fc9833dc2df]
[bt]: (72) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgio-2.0.so.0(g_application_register+0x112) [0x7fc9835234d2]
[bt]: (73) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgio-2.0.so.0(+0xe98f0) [0x7fc9835238f0]
[bt]: (74) /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libgio-2.0.so.0(g_application_run+0x106) [0x7fc983523c46]
[bt]: (75) /etc/profiles/per-user/james/bin/xournalpp(_ZN11XournalMain3runEiPPc+0x636) [0x4d5846]
[bt]: (76) /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(+0x2a10e) [0x7fc982a3d10e]
[bt]: (77) /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(__libc_start_main+0x89) [0x7fc982a3d1c9]
[bt]: (78) /etc/profiles/per-user/james/bin/xournalpp(_start+0x25) [0x4cf755]


Try to get a better stacktrace...
[bt] #0 /etc/profiles/per-user/james/bin/xournalpp() [0x4d0f98]
?? ??:0
[bt] #1 /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(+0x3ff30) [0x7fc982a52f30]
__restore_rt at ??:?
[bt] #2 /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(+0x8fefc) [0x7fc982aa2efc]
__pthread_kill_implementation at ??:?
[bt] #3 /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(gsignal+0x16) [0x7fc982a52e86]
__GI_raise at ??:?
[bt] #4 /nix/store/apab5i73dqa09wx0q27b6fbhd1r18ihl-glibc-2.39-31/lib/libc.so.6(abort+0xd7) [0x7fc982a3b935]
__GI_abort at ??:?
[bt] #5 /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libglib-2.0.so.0(+0x23015) [0x7fc983281015]
g_assertion_message_expr.cold at ??:?
[bt] #6 /nix/store/bcms6vbac6jyffwyj8y630brpy1il4ks-glib-2.80.0/lib/libglib-2.0.so.0(g_assertion_message_error+0xb1) [0x7fc9832e9d91]
g_assertion_message_error at ??:?
[bt] #7 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1cb850) [0x7fc9839cb850]
ensure_surface_for_gicon at ??:?
[bt] #8 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1cbd27) [0x7fc9839cbd27]
gtk_icon_helper_load_surface at ??:?
[bt] #9 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1cc03e) [0x7fc9839cc03e]
_gtk_icon_helper_get_size at ??:?
[bt] #10 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1e019d) [0x7fc9839e019d]
gtk_image_get_content_size at ??:?
[bt] #11 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
gtk_css_custom_gadget_get_preferred_size at ??:?
[bt] #12 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
gtk_css_gadget_get_preferred_size at ??:?
[bt] #13 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x1e08c3) [0x7fc9839e08c3]
gtk_image_get_preferred_height_and_baseline_for_width at ??:?
[bt] #14 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4863) [0x7fc983aa4863]
gtk_widget_query_size_for_orientation at ??:?
[bt] #15 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
gtk_widget_get_preferred_height_and_baseline_for_width at ??:?
[bt] #16 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
gtk_css_custom_gadget_get_preferred_size at ??:?
[bt] #17 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
gtk_css_gadget_get_preferred_size at ??:?
[bt] #18 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xe74d3) [0x7fc9838e74d3]
gtk_button_get_preferred_height_and_baseline_for_width at ??:?
[bt] #19 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4863) [0x7fc983aa4863]
gtk_widget_query_size_for_orientation at ??:?
[bt] #20 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
gtk_widget_get_preferred_height_and_baseline_for_width at ??:?
[bt] #21 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xd9c48) [0x7fc9838d9c48]
gtk_box_get_content_size at ??:?
[bt] #22 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x12dc03) [0x7fc98392dc03]
gtk_css_custom_gadget_get_preferred_size at ??:?
[bt] #23 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x132fb8) [0x7fc983932fb8]
gtk_css_gadget_get_preferred_size at ??:?
[bt] #24 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0xd8fa3) [0x7fc9838d8fa3]
gtk_box_get_preferred_height_and_baseline_for_width at ??:?
[bt] #25 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4863) [0x7fc983aa4863]
gtk_widget_query_size_for_orientation at ??:?
[bt] #26 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height+0x5c) [0x7fc983aa407c]
gtk_widget_get_preferred_height at ??:?
[bt] #27 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a4746) [0x7fc983aa4746]
gtk_widget_query_size_for_orientation at ??:?
[bt] #28 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0x7a) [0x7fc983aa4f2a]
gtk_widget_get_preferred_height_and_baseline_for_width at ??:?
[bt] #29 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x2a507b) [0x7fc983aa507b]
_gtk_widget_get_preferred_size_and_baseline at ??:?
[bt] #30 /nix/store/nx27cp48ylj2bmrf1r72ncl3pcq7w5df-gtk+3-3.24.41/lib/libgtk-3.so.0(+0x30404c) [0x7fc983b0404c]
gtk_toolbar_measure at ??:?


Execution log:

DEBUG   : GLib-GIO :: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
DEBUG   : GLib-GIO :: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
DEBUG   : GLib-GIO :: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
DEBUG   : GLib :: unsetenv() is not thread-safe and should not be used after threads are created
MESSAGE : xopp :: TEXTDOMAINDIR = (null), Platform-specific locale dir = /nix/store/mi22mvji6xqnmxsl31c7id297imhvgww-xournalpp-1.2.3/share/xournalpp/../locale, chosen directory = /nix/store/mi22mvji6xqnmxsl31c7id297imhvgww-xournalpp-1.2.3/share/xournalpp/../locale
WARNING : xopp :: XournalMain: System default locale could not be set.
 - Caused by: locale::facet::_S_create_c_locale name not valid
 - Note that it is not supported to set the locale using mingw-w64 on windows.
 - This could be solved by compiling xournalpp with msvc
INFO    : xopp :: Loading plugins from: /nix/store/mi22mvji6xqnmxsl31c7id297imhvgww-xournalpp-1.2.3/share/xournalpp/plugins
INFO    : xopp :: Loading plugins from: /home/james/.config/xournalpp/plugins
MESSAGE : xopp :: Plugin "MigrateFontSizes" UI initialized
WARNING : xopp :: [Crash Handler] Crashed with signal 6
WARNING : xopp :: [Crash Handler] Wrote crash log to: /home/james/.cache/xournalpp/errorlogs/errorlog.20240517-151326.log
```

**Comment by rolandlo** [comment]
It's probably not related to the problem, but let me note, that the warning
```term
WARNING : xopp :: XournalMain: System default locale could not be set.
 - Caused by: locale::facet::_S_create_c_locale name not valid
 - Note that it is not supported to set the locale using mingw-w64 on windows.
 - This could be solved by compiling xournalpp with msvc
```
is not very helpful on Linux.

**Comment by RandomSpaceship** [comment]
I can't run the AppImage file at all, in X11 or Wayland. Trying to run it in Dolphin and zsh both error with `No such file or directory`, running it in bash gives `bash: ./xournalpp.AppImage: cannot execute: required file not found`, and running it with [appimage-run](https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-appimageTools) gives the following error log:
```
xournalpp.AppImage installed in /home/james/.cache/appimage-run/99c2e385cdfef2eb59ac367214fc37c31855779e0910990b0f1612d7ad339f8b
/home/james/.cache/appimage-run/99c2e385cdfef2eb59ac367214fc37c31855779e0910990b0f1612d7ad339f8b/AppRun.wrapped: error while loading shared libraries: libthai.so.0: cannot open shared object file: No such file or directory
```
Which persists regardless of whether or not `libthai` is actually installed.

However, when I ran the `xournalpp` command from the terminal, it gave a very different (and much more useful) error log:
```
(com.github.xournalpp.xournalpp:15615): Gtk-WARNING **: 17:47:51.685: Locale not supported by C library.
        Using the fallback 'C' locale.
xopp-Message: 17:47:52.007: TEXTDOMAINDIR = (null), Platform-specific locale dir = /nix/store/mi22mvji6xqnmxsl31c7id297imhvgww-xournalpp-1.2.3/share/xournalpp/../locale, chosen directory = /nix/store/mi22mvji6xqnmxsl31c7id297imhvgww-xournalpp-1.2.3/share/xournalpp/../locale

(com.github.xournalpp.xournalpp:15615): xopp-WARNING **: 17:47:52.007: XournalMain: System default locale could not be set.
 - Caused by: locale::facet::_S_create_c_locale name not valid
 - Note that it is not supported to set the locale using mingw-w64 on windows.
 - This could be solved by compiling xournalpp with msvc
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.front
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround21
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround21
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround40
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround41
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround50
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround51
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.surround71
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.iec958
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.hdmi
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.hdmi
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.modem
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.modem
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.phoneline
ALSA lib pcm.c:2792:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.phoneline
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
xopp-Message: 17:47:52.153: Plugin "MigrateFontSizes" UI initialized
**
Gtk:ERROR:../gtk/gtkiconhelper.c:495:ensure_surface_for_gicon: assertion failed (error == NULL): Icon 'image-missing' not present in theme Adwaita (gtk-icon-theme-error-quark, 0)
Bail out! Gtk:ERROR:../gtk/gtkiconhelper.c:495:ensure_surface_for_gicon: assertion failed (error == NULL): Icon 'image-missing' not present in theme Adwaita (gtk-icon-theme-error-quark, 0)

(com.github.xournalpp.xournalpp:15615): xopp-WARNING **: 17:47:52.211: [Crash Handler] Crashed with signal 6

(com.github.xournalpp.xournalpp:15615): xopp-WARNING **: 17:47:52.211: [Crash Handler] Wrote crash log to: /home/james/.cache/xournalpp/errorlogs/errorlog.20240522-174752.log
/nix/store/m3gq2f6iwh3g67zrs27m7w82b37l8gf8-binutils-wrapper-2.41/bin/addr2line: 'xournalpp': No such file

(com.github.xournalpp.xournalpp:15615): xopp-WARNING **: 17:47:52.319: Trying to emergency save the current open document?

(com.github.xournalpp.xournalpp:15615): xopp-WARNING **: 17:47:52.319: Successfully saved document to "/home/james/.config/xournalpp/emergencysave.xopp"
```
Since it complained about an assertion failure finding Adwaita icons, I installed the `gnome.adwaita-icon-theme` package, and it runs now - and even respects my system zoom level! I still can't get the AppImage running though weirdly enough, even though the main executable is happy. I've done a quick test and it all seems to be working normally (pressure sensitivity, palm rejection, stylus buttons).

**TLDR:** Running Xournal++ from the command line gave a more useful error, it was failing to find Adwaita icons. Installing that package made the installed executable run, even though the AppImage fails to for other reasons.

**Comment by RandomSpaceship** [comment]
With a proper error message to search, I've found that it appears to be already reported on the `nixpkgs` side [here](https://github.com/NixOS/nixpkgs/issues/163107). No idea what the `libthai` stuff is all about, but it seems that the icons issue is an artifact of something on the NixOS side, and actually has nothing to do with Xournal++ code.

**Comment by Korne127** [comment]
At least the AppImage part of this should be fixed with https://github.com/AppImageCommunity/pkg2appimage/pull/555
